runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -489,21 +489,28 @@ end
 
 --- Request workspace-wide diagnostics.
 --- @param opts vim.lsp.WorkspaceDiagnosticsOpts
-function M._workspace_diagnostics(opts)
+--- @param callback? fun(errors: lsp.ResponseError[]) Function called after the request is completed.
+function M._workspace_diagnostics(opts, callback)
   local clients = lsp.get_clients({ method = ms.workspace_diagnostic, id = opts.client_id })
+  local remaining = #clients
+  local errors = {} --- @type lsp.ResponseError[]
```

Comment:
This should probably a map of client id to error.


```suggestion
  local errors = {} --- @type table<integer,lsp.ResponseError?>
```